### PR TITLE
Add publish to maven central task

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add `publish` task that is configured to publish to MavenCentral.

### Why

This is to automate the MavenCentral publication to minimize human errors.

### Known limitations

Pending verification by the SonaType OSSRH team.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 [Anchor Platform](anchor-platform.md): The anchor platform Spring Boot application.
 
+# Publishing 
+[How to publish](publishing.md)  `stellar-anchor-sdk.jar` from `core` subproject.
```
Comment (marcelosalloum) [inline docs/CONTRIBUTING.md:9]: 💡  Do you mean we should execute the jar?

```suggestion
[How to publish](publishing.md)  Execute `stellar-anchor-sdk.jar` from `core` subproject.
```

File: docs/publishing.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+3. If `publish` task was executed successfully, we should see the repository successfully publishe.
+   Open the [OSSRH Console](https://oss.sonatype.org) to manage the publication.
+
+4. Under `Staging Repositories`, select the published repository.
+5. Click on `Drop` to delete the publication.
+6. Click on `Close` to trigger the MavenCentral validation process. The sync process of the artifactory to 
+   MavenCentral is enabled by [OSSRH-78005](https://issues.sonatype.org/browse/OSSRH-78005) ticket.  
+
+# Verify the Publication
+The repository should be searchable at: [Maven Search](https://search.maven.org/search?q=org.stellar).
```
Comment (marcelosalloum) [inline docs/publishing.md:63]: ❓  I only found `core` in maven. Will we publish the remaining ones later?

File: core/build.gradle.kts [hunk 3]
```diff
@@ -27,3 +29,69 @@ dependencies {
...
+                        }
+                        developer {
+                            id.set("JakeUrban")
+                            name.set("Jake Urban")
+                            email.set("jake@stellar.org")
+                        }
+                        developer {
+                            id.set("marcelosalloum")
+                            name.set("Marcelo Salloum")
+                            email.set("jake@stellar.org")
```
Comment (marcelosalloum) [inline core/build.gradle.kts:72]: 👀  my email is wrong:
```suggestion
                            email.set("marcelo@stellar.org")
```

File: docs/publishing.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+# Import the Stellar Anchor Java SDK
+
+## Maven
+```
+<dependency>
+    <groupId>org.stellar</groupId>
+    <artifactId>stellar-anchor-sdk</artifactId>
+    <version>${version}</version>
+</dependency>
```
Comment (marcelosalloum) [inline docs/publishing.md:73]: 🤔  Should we have some kind of way to select only specific modules? Like `<artifactId>stellar-anchor-sdk:core</artifactId>` for instance?

**Review by marcelosalloum — COMMENTED** [review]
Found a wrong email 👀  and left a few questions below.

Also, Github Actions allow you to publish to maven using [this config](https://docs.github.com/en/actions/publishing-packages/publishing-java-packages-with-maven#publishing-packages-to-the-maven-central-repository). It's similar to what we have [in the js-stellar-sdk](https://github.com/stellar/js-stellar-sdk/blob/master/.github/workflows/npm_publish.yml) to publish packages to npm every time we create a [GH release](https://github.com/stellar/js-stellar-sdk/releases). Maybe a nice-to-have for a future PR? I'd be happy to help with that later.

File: core/build.gradle.kts (hunk 3 above)
Comment (lijamie98) [reply core/build.gradle.kts]: Cool catch!!!

File: docs/publishing.md (hunk 4 above)
Comment (lijamie98) [reply docs/publishing.md:73]: I changed the groupId for better name spacing as this:
```
<dependency>
    <groupId>org.stellar.anchor-sdk</groupId>
    <artifactId>core</artifactId>
    <version>${version}</version>
</dependency>
```

File: build.gradle.kts [hunk 5]
```diff
@@ -7,6 +7,9 @@ subprojects {
   apply(plugin = "java")
   apply(plugin = "com.diffplug.spotless")
 
+  group = "org.stellar"
+  version = "1.0.1"
```
Comment (JakeUrban) [inline build.gradle.kts]: Are we sure we want to publish under v1? That’s communicating that we won’t break the interface in any other v1 release. Maybe we use v0.1? And release 1.0 when we know our interface is solid. Up to you though.

**Review by JakeUrban — APPROVED** [review]

File: build.gradle.kts (hunk 5 above)
Comment (lijamie98) [reply build.gradle.kts]: Just renamed the version as we speak. The version is changed to 0.1.0

**Pull request merged**